### PR TITLE
added dependencies, reduced size of example notebooks, fixed euclidean_distance function, added explore_features to init, changed some preprocessing methods to be called with df instead of dict, changed compute_direction and compute_turning functions by i.a. adding function cosine_similarity, changed some functions (f.e. get_heading_difference or outlier_detection) due to changes in how direction is calculated, added functions calculate_direction_angle and calculate_turning_angle for 2D data, changed the sorting order in regrouping data, fixed how average_acceleration is calculated, replaced unit test for old clustering function with unit test for new clustering function, added labels to axes in returned dataframes in both ts_feature functions, changed multiple unit tests which failed due to new functions or other reasons

### PR DESCRIPTION
**Pull request by TimKleinlein**

### CODE REVIEW

**Review by eren-ck — COMMENTED** [review]
One question I have is why are you not using `sklearn.metrics.pairwise.cosine_similarity` or the `scipy.spatial.distance.cosine` methods to compute the cosine similarity?

**Comment by TimKleinlein** [comment]
you're right that makes more sense, I change that.

**Review by eren-ck — APPROVED** [review]
